spec/javascripts/components/intervention-spec.js

```diff
@@ -0,0 +1,91 @@
+    container = document.createElement('div')
+    container.innerHTML =
+      '<section class="gem-c-intervention" data-module="intervention" data-intervention-name="test-intervention-name"><a class="govuk-link gem-c-intervention__dismiss-link">Dismiss</a></section>'
+
+    document.body.appendChild(container)
+    var element = document.querySelector('[data-module="intervention"]')
+    new GOVUK.Modules.Intervention(element).init()
+
+    window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+  })
+
+  afterEach(function () {
+    GOVUK.deleteCookie('intervention_campaign')
+    document.body.removeChild(container)
+  })
+
+  describe('close banner', function () {
+    it('should hide intervention banner', function () {
+      var banner = document.querySelector('.gem-c-intervention')
+      var close = document.querySelector('.gem-c-intervention__dismiss-link')
+      expect(banner).toBeVisible()
+
+      close.click()
+
+      expect(banner).toBeHidden()
+    })
+
+    it('sets a cookie value', function () {
+      spyOn(GOVUK, 'cookie').and.callThrough()
+      var close = document.querySelector('.gem-c-intervention__dismiss-link')
+      close.click()
+
+      expect(GOVUK.cookie).toHaveBeenCalled()
+      var bannerCookie = GOVUK.cookie('intervention_campaign')
+      expect(bannerCookie).toEqual('test-intervention-name')
+    })
+  })
+
+  describe('cookies are already set', function () {
+    it('does not display the banner', function () {
+      GOVUK.setCookie('intervention_campaign', 'test-intervention-name', { days: 1 })
+      var element = document.querySelector('[data-module="intervention"]')
+      new GOVUK.Modules.Intervention(element).init()
+
+      var banner = document.querySelector('.gem-c-intervention')
+
+      expect(banner).toBeHidden()
+    })
+  })
+})
+
+describe('another concurrent banner campaign', function () {
+  'use strict'
+
+  var container
+
+  beforeEach(function () {
+    GOVUK.setCookie('intervention_campaign', 'existing-value', { days: 1 })
+    window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+
+    container = document.createElement('div')
+    container.innerHTML =
+      '<section class="gem-c-intervention" data-module="intervention" data-intervention-name="test-intervention-name"><a class="govuk-link gem-c-intervention__dismiss-link">Dismiss</a></section>'
+
+    document.body.appendChild(container)
+    var element = document.querySelector('[data-module="intervention"]')
+    new GOVUK.Modules.Intervention(element).init()
+  })
+
+  it('appends campaign value to the cookie', function () {
+    var close = document.querySelector('.gem-c-intervention__dismiss-link')
+    close.click()
+
+    var updatedCookie = GOVUK.cookie('intervention_campaign')
+    expect(updatedCookie).toEqual('existing-value,test-intervention-name')
+  })
+
+  afterEach(function () {
```

Comment:
Nitpick: normally we include the `afterEach` function immediately after the `beforeEach` for readability.